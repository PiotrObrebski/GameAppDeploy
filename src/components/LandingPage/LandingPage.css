.landing-page{
  width: 80%;
  max-width: 1024px;
}
