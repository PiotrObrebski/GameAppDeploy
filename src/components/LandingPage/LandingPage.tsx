import './LandingPage.css';

import React, { ReactElement, useEffect } from 'react';
import { connect } from 'react-redux';

import { GAMES_END_POINT, USER_END_POINT } from '../../api-end-points';
import { getDataFromServer } from '../../api-helper';
import { setAllGames, setUser } from '../../redux/actions';
import { GamesList } from '../GamesList/GamesList';

const LandingPage = (
  props: {
    setUser: (content: User) => void;
    setAllGames: (content: IGame[]) => void;
    user: User;
    allGames: {
      gamesArray: IGame[]
    }
  }
): ReactElement => {  
  useEffect(() => {
    getDataFromServer(USER_END_POINT)
    .then( response => {
      props.setUser(response)
    })
    
    getDataFromServer(GAMES_END_POINT)
    .then( response => {
      props.setAllGames(response)
    })
  }, []);

  if(props.allGames.gamesArray[0]?.id.length > 0 && props.user.id.length > 0 )
  {
    return (
      <div className="landing-page">
        <h1>Hello {props.user.name}</h1>
        <div>What would you like to play today?</div>
        <GamesList allGames={props.allGames}/>
      </div>
      )
  } else {
    return (
      <div className="landing-page"></div>
    )
  }
};

const mapStateToProps = (
  state: {
    user: User;
    allGames: {
      gamesArray: IGame[];
    }
  }
) => {
  const { user, allGames } = state;
  return { user, allGames };
}

export default connect(
  mapStateToProps, 
  {
    setUser,
    setAllGames
  }
)(LandingPage);