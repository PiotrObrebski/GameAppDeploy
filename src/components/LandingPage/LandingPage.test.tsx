import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import store from '../../redux/store';
import LandingPage from './LandingPage';

describe("LandingPage", () => {
  let component: ShallowWrapper<{ any: any }>;

  beforeEach(() => {
    component = shallow(
      <Provider store={store}>
        <LandingPage />
      </Provider>);
  });

  test("Should mount", () => {
    expect(component.length).toBe(1);
  });

  test("should match snapshot", () => {
    const LandingPageRender = renderer
      .create(
        <Provider store={store}>
          <LandingPage />
        </Provider>
      )
      .toJSON();
    expect(LandingPageRender).toMatchSnapshot();
  });
});
