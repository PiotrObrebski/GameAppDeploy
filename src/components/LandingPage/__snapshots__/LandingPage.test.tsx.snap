// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match snapshot 1`] = `
<div
  className="landing-page"
/>
`;
